Jropp/jason-ropp-js-iobp
# Onboarding Project: User-Component Upgrade

## What It Does

Combines the functionality of, create, display and edit-user into a single `<user-component>` element. 

Prior to this, edit user, create user, and user profile were all their own html code on separate pages. Now the `<user-component>` card takes in three possible attributes

```
<user-component mode="[create, display, edit]" collapsed="[true, false]" user="{{userObject}}">
```

There are three dom-if templates displayed by switching boolean properties. These boolean properties are controlled from the parent via the `mode="[create, display, edit]"` attribute. `The collapsed=["true/false"]` attribute is a dom-if template inside of the display card that hides the details if set to true. Clicking a collapsed card will expand it and show the details and the edit button. This feature is for collapsing the card for users list.

This addresses #32 

## How To Test

Click the create user tab. Fill out the form for a new user. Notice that if you leave a field blank The user will not save. And you should be notified via window alert. When you create the user properly, the form fields will reset. 

If you click on the user list tab your newly created user should be displayed.

To view the details of that user, click on the card, it will expand to show you the details, including an autogenerated unique id that follows that user. Click edit user and alter the data, then click submit. You should notice changes in the displayed user card immediately. 

From the edit user window you can also delete that created user. 

At this point, clicking the user profile tab just automatically displays the first user in the database. This will be easily fixed once the parent component is another custom element so that the current user can be data bound as an attribute in <user-component>

## Notes/Caveats

- This is still using local storage database. This will be replaced by API calls to a database in coming PRs

- This is still implemented for Chrome only. This will be fixed in a future PR via #38 

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:

## Dependencies

- [ ] #30
